src/main/java/com/laborete/LaboreteAPI/profile/services/UsersServiceImpl.java

```diff
@@ -36,33 +43,53 @@ public class UsersServiceImpl implements UsersService {
     private static final String FILE_IS_EMPTY = "The file cannot be empty";
     private static final String FILE_NAME_IS_EMPTY = "The file name cannot be empty";
     private static final String FILE_FAILED_UPLOAD = "Failed to upload file: ";
+
+    private static final String FAILED_TO_EXTRACT_BYTES = "Failed to extract bytes";
+    private static final String AVATAR_NOT_FOUND = "Avatar was not found";
+
+    private static final String BACKGROUND_NOT_FOUND = "Background was not found";
+    private static final String DIRECTORY_NOT_FOUND = "Directory was not found";
+
     private static final String ERROR_CREATING_DIRECTORY = "The directory was not created";
     private final UsersRepository usersRepository;
     private final UserAvatarRepository userAvatarRepository;
+    private final UserBackgroundRepository userBackgroundRepository;
     private final UserMapper userMapper;
-    @Value("${directory.path}")
-    private String ROOT_PATH;
+
+
+    @Value("${directory.path.avatars}")
+    private String ROOT_PATH_AVATARS;
+    @Value("${directory.path.background}")
+    private String ROOT_PATH_BACKGROUND;
+
 
     public UsersServiceImpl(
             UsersRepository usersRepository,
             UserAvatarRepository userAvatarRepository,
-            UserMapper userMapper
-    ) {
+            UserMapper userMapper,
+            UserBackgroundRepository userBackgroundRepository) {
         this.usersRepository = usersRepository;
         this.userAvatarRepository = userAvatarRepository;
         this.userMapper = userMapper;
+        this.userBackgroundRepository = userBackgroundRepository;
     }
 
+    @SneakyThrows
```

Comment:
SneakyThrows is a bad practice! We only should use this attribute in unit tests. Please get rid of it.